javascript/javascript3/week1/weatherApp/weatherApp.js

```diff
@@ -0,0 +1,74 @@
+
+const inputElement = document.getElementById('input-city');
+const btnElement = document.getElementById('weatherButton');
+
+btnElement.addEventListener('click', ()=> {
+
+    if(inputElement.value){
+        const cityName = inputElement.value.toLowerCase();
+        let key = '6284b3975ab711a2f7c25e4a1d0434a9';
+        const api = `https://api.openweathermap.org/data/2.5/weather?q=${cityName}&APPID=${key}`;
+
+        fetch(api)
+        .then(res => res.json())
+        .then(data => displayWeather(data));
+    }
+    else {
+        alert('Enter the city name ?');  
+    }
+});
+function displayWeather(data){
+    
+    const iconElement = document.querySelector(".weather-icon");
+    iconElement.innerHTML = `<img src="icons/${data.weather[0].icon}.png"/>`;
+
+    const tempElement = document.querySelector(".temperature-value p");
+    tempElement.innerHTML = `${Math.floor(data.main.temp - 273)}°<span>C</span>`;
+
+    const descElement = document.querySelector(".temperature-description p");
+    descElement.innerHTML = `Description: ${data.weather[0].description}`;
+
+    const locationElement = document.querySelector(".location p");
+    locationElement.innerHTML = `Location: ${data.name}, ${data.sys.country}`;
+
+    const windElement = document.querySelector(".wind-speed p");
+    windElement.innerHTML = `Wind Speed : ${data.wind.speed}`;
+
+    const cloudyElement = document.querySelector(".cloudy p");
+    cloudyElement.innerHTML = `How cloudy is : ${data.clouds.all}`;
+
+    const sunRiseSetElement = document.querySelector(".sun-rise-set p");
+    sunRiseSetElement.innerHTML = `Sun rise ${data.sys.sunrise}, Sun set ${data.sys.sunset}`;    
+}
+function getWeatherByGeoLocation(lat,lon){
+        key = '6284b3975ab711a2f7c25e4a1d0434a9';
+        let url = `http://api.openweathermap.org/data/2.5/weather?lat=${lat}&lon=${lon}&appid=${key}`;
+        
+        fetch(url)
+            .then(res => res.json())
+            .then((data) => displayWeather(data));
```

Comment:
Since you have already used (almost) the same lines above when you did the other api call, would it be possible to make a function that could used in both places, instead of having to write almost the same thing two times?

Reply:
yes Noted 
thanks